src/Menu/CachedFrontendMenuBuilder.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bolt\Menu;
+
+use Bolt\Collection\DeepCollection;
+use Psr\SimpleCache\CacheInterface;
+use Symfony\Component\Stopwatch\Stopwatch;
+
+class CachedFrontendMenuBuilder
+{
+    /** @var CacheInterface */
+    private $cache;
+
+    /** @var FrontendMenuBuilder */
+    private $menuBuilder;
+
+    /** @var Stopwatch */
+    private $stopwatch;
+
+    public function __construct(CacheInterface $cache, FrontendMenuBuilder $menuBuilder, Stopwatch $stopwatch)
+    {
+        $this->cache = $cache;
+        $this->menuBuilder = $menuBuilder;
+        $this->stopwatch = $stopwatch;
+    }
+
+    public function getMenu(?string $name = ''): ?DeepCollection
```

Comment:
why backend returns string while frontend returns collection?

Reply:
Backend uses KNP-menu internally, but we only output a string-as-JSON for Vue.

Frontend is a collection, that can be used in the implementor's own code. 

It's what we discussed the other day: Either we move _both_ to KNP, or we move _both_ out of it, but we didn't decide which yet. :-)

Reply:
Or we move the handler out of builder, so it will become knp-agnostic?

Reply:
And eventually you will have to output json in frontend as well, for headless.

Reply:
I do think it'd be good to use a similar concept for both.. Let's: 

 - Convert the backend to use `DeepCollection`, instead of KNP
 - Add a `getJsonMenu()` method to both, so they have the same 'interface'.

Reply:
Keeping a collection is M part.
Showing it in json format is V part.